src/main/scala/info/bethard/anafora/Anafora.scala

```diff
@@ -39,9 +45,12 @@ class Entity(xml: Elem) extends Annotation(xml) {
       case Array(start, end) => (start, end)
     }).sorted
   lazy val fullSpan: (Int, Int) = (spans.map(_._1).min, spans.map(_._2).max)
-  def text(implicit data: Data): String = spans.map{
-    case (start, end) => data.text.substring(start, end)
-  }.mkString("...")
+  def text(implicit data: Data): Option[String] = data.text match {
```

Comment:
I believe this could be more concisely written as:
```
data.text.map(text => spans.map {
  case (start, end) => text.substring(start, end)
}.mkString("..."))
```